app/windows/codesign.js

```diff
@@ -0,0 +1,166 @@
+const { exec } = require('child_process');
+const path = require('path');
+const os = require('os');
+const fs = require('fs');
+
+const SIGN_JSON_TEMPLATE = {
+  Version: '1.0.0',
+  DriEmail: ['joaquim.rocha@microsoft.com'],
+  GroupId: null,
+  CorrelationVector: null,
+  SignBatches: [],
+};
+
+const POLICY_JSON = {
+  Version: '1.0.0',
+  Intent: '',
+  ContentType: '',
+  ContentOrigin: '',
+  ProductState: '',
+  Audience: '',
+};
+
+const AUTH_JSON = {
+  Version: '1.0.0',
+  AuthenticationType: 'AAD_CERT',
+  ClientId: `${process.env.HEADLAMP_WINDOWS_CLIENT_ID}`,
+  AuthCert: {
+    SubjectName: `CN=${process.env.HEADLAMP_WINDOWS_CLIENT_ID}.microsoft.com`,
+    StoreLocation: 'LocalMachine',
+    StoreName: 'My',
+    SendX5c: 'true',
+  },
+  RequestSigningCert: {
+    SubjectName: `CN=${process.env.HEADLAMP_WINDOWS_CLIENT_ID}`,
+    StoreLocation: 'LocalMachine',
+    StoreName: 'My',
+  },
+};
+
+function getFileList(rootDir) {
+  let files = {};
+  let dirs = ['.'];
+  while (dirs.length > 0) {
+    const dirName = dirs.shift();
+    const curDir = path.join(rootDir, dirName);
+
+    fs.readdirSync(curDir).forEach(file => {
+      if (file === 'node_modules') {
+        return;
+      }
+      const filepath = path.resolve(rootDir, dirName, file);
+      const stat = fs.statSync(filepath);
+      if (stat.isDirectory() && !files[file]) {
+        dirs.push(path.join(dirName, file));
+        files[file] = [];
+      } else {
+        if (!files[dirName]) {
+          files[dirName] = [];
+        }
+
+        files[dirName].push(file);
+      }
+    });
+  }
+  return files;
+}
+
+function createJson(pathToSign, fileName = 'test_SignInput.json') {
```

Comment:
Can we include what type of json structure in the name? Or add a function comment block to describe it.

Also, can you please include a URL to the docs for this json structure?

Reply:
There are no public docs to this. So I will add them in our internal guides to building signed binaries.

Reply:
Cool thanks. I'll have a look there.